serverv2/cometbft/params.go

```diff
@@ -0,0 +1,26 @@
+package cometbft
+
+import (
+	"context"
+
+	cmtproto "github.com/cometbft/cometbft/proto/tendermint/types"
+)
+
+const InitialAppVersion uint64 = 0
+
+// ParamStore defines the interface the parameter store used by the BaseApp must
+// fulfill.
+type ParamStore interface {
+	Get(ctx context.Context) (cmtproto.ConsensusParams, error)
+	Has(ctx context.Context) (bool, error)
+	Set(ctx context.Context, cp cmtproto.ConsensusParams) error
+}
+
+// AppVersionModifier defines the interface fulfilled by BaseApp
+// which allows getting and setting it's appVersion field. This
+// in turn updates the consensus params that are sent to the
+// consensus engine in EndBlock
+type AppVersionModifier interface {
+	SetAppVersion(context.Context, uint64) error
+	AppVersion(context.Context) (uint64, error)
+}
```

Comment:
this should live in the appmanger, imo since that is where we will be handling upgrades